src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/Snapshots/ImmutablePropertyCollection.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    /// <summary>
+    ///     Abstract immutable collection that supports lookup by index and name.
+    /// </summary>
+    internal abstract class ImmutablePropertyCollection<T> : IEnumerable where T : class
```

Comment:
Any reason not to implement `IEnumerable<T>` instead?

Reply:
Extra implementation that will never be called; the `IVsXXX` collections only use IEnumerable.